### PR TITLE
build: search first in CMAKE_INSTALL_PREFIX

### PR DESCRIPTION
**Pull request by tomereli**
- When dependent libraries are used by the framework (currently only
MSGLIB - and specifically nng) - first search for them in
CMAKE_INSTALL_PREFIX. With this, the framework will first look for nng in the default
installation folder (multiap/install/) when building with maptools, and
will find nng there is it was built.

- update RPATH so that libraries found in CMAKE_INSTALL_PREFIX will always be used first.

Signed-off-by: Tomer Eliyahu <tomer.b.eliyahu@intel.com>

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -49,9 +49,15 @@ endif()
...
 if (TARGET_PLATFORM STREQUAL "rdkb")
-        set(CMAKE_SKIP_RPATH TRUE)
+    set(CMAKE_SKIP_RPATH TRUE)
 endif()
 
+list (FIND CMAKE_PLATFORM_IMPLICIT_LINK_DIRECTORIES "${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_LIBDIR}" isSystemDir)
+if ("${isSystemDir}" STREQUAL "-1")
+    set (CMAKE_INSTALL_RPATH "${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_LIBDIR}")
+endif ("${isSystemDir}" STREQUAL "-1")
+set (CMAKE_INSTALL_RPATH_USE_LINK_PATH TRUE)
```
Comment (arnout) [inline CMakeLists.txt]: According to the "Common questions" section in the referenced wiki page, this one line should be enough. And I think that also [`CMAKE_INSTALL_RPATH_USE_LINK_PATH`](https://cmake.org/cmake/help/v3.0/variable/CMAKE_BUILD_WITH_INSTALL_RPATH.html?highlight=cmake_build_with_install_rpath) needs to be set to TRUE. But I don't think all the rest is needed.

**Comment by arnout** [comment]
I really like this
```
Dependencies - None
Tests - ...
```
in the commit message. Should be added to our guidelines of how to write a commit message.

File: CMakeLists.txt (hunk 1 above)
Comment (tomereli) [reply CMakeLists.txt]: I tried setting only `CMAKE_INSTALL_RPATH_USE_LINK_PATH` and got 
```
]$ ./build/install/bin/tests/common/poller_test 
./build/install/bin/tests/common/poller_test: error while loading shared libraries: mapfcommon.so.1: cannot open shared object file: No such file or directory
```

So I guess it is needed...

Also, see nng which has a pretty decent cmake support, this is done the same there: https://github.com/nanomsg/nng/blob/master/CMakeLists.txt

**Comment by tomereli** [comment]
> I really like this
> 
> ```
> Dependencies - None
> Tests - ...
> ```
> 
> in the commit message. Should be added to our guidelines of how to write a commit message.

Agree, I already told my team members, this should be in CONTRIBUTERS.md I suppose?

**Comment by arnout** [comment]
>> I really like this
>>
>> ```
>> Dependencies - None
>> Tests - ...
>> ```
>>
>> in the commit message. Should be added to our guidelines of how to write a commit message.
> 
> Agree, I already told my team members, this should be in CONTRIBUTERS.md I suppose?

CONTRIBUTING.md or Contributing.md.

Should be in the merged repository, of course.

**Pull request merged**